### PR TITLE
fix(animation): remove throw and fix promise handling

### PR DESCRIPTION
**Pull request by NathanWalker**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [ ] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

## What is the current behavior?

Usage of `throw` destabilizes apps during animation handling.

## What is the new behavior?

Promise rejection is now removed around keyframe handling since there was no valid way to handle. With tracing enabled developers can now have meaningful and productive debug sessions.

https://github.com/NativeScript/nativescript-angular/issues/431

### CODE REVIEW

**Comment by NathanWalker** [comment]
Excellent @RoyiNamir definitely agree. I’ll update this PR to include that fix as well later, thank you.

**Comment by NathanWalker** [comment]
@RoyiNamir that spot is handled in this pr.

**Comment by NathanWalker** [comment]
@RoyiNamir in your case above, anytime you use animation.play() it'd be best practice to wire up the success and error callbacks, ie:

```
animation.play(options).then(success => {
  //...
}, error => {
  //...
});
```

**Comment by RoyiNamir** [comment]
@NathanWalker Not only. In my case ^ it was when I added a class ( which contains as single `animation-name` with 1 iteration) and then removed that class.
Error appeared also.

**Comment by NathanWalker** [comment]
@RoyiNamir good to know. Nice one and valid use case. Will update PR to handle that as well, thank you for clarifying.

**Comment by NathanWalker** [comment]
@RoyiNamir There really should never be a rejection in keyframe animation handling since there's no proper way to handle it therefore this removes rejections altogether. Will solve your case and many others that have been reported over time around this. You may try the changes seen in this PR in your project and report back the results for you.

**Comment by RoyiNamir** [comment]
@NathanWalker  Having said that , there are also much more critical animation problems which is in progress of solving by the NS team.
I've already reported a memory leak which eventually stuck and shutdown the app. 
https://github.com/NativeScript/NativeScript/issues/5731

Just to clarify : I'm not saying that the current issue is not important  , as it is just an informative error.
You can't do `addAnimationClass().then(()=>{},error=>do somethng)`
And as you've said , it is feasible to do `view.animate()).then(()=>{},error=>do somethng)`

**Comment by vakrilov** [comment]
test

File: tns-core-modules/ui/animation/keyframe-animation.ts [hunk 1]
```diff
@@ -239,10 +236,10 @@ export class KeyframeAnimation implements KeyframeAnimationDefinition {
             // Catch the animation cancel to prevent unhandled promise rejection warnings
             animation.play().then(() => {
                 this.animate(view, index + 1, iterations);
+            }, (error: any) => {
+                traceWrite(typeof error === "string" ? error : error.message, traceCategories.Animation, traceType.warn);
```
Comment (vakrilov) [inline tns-core-modules/ui/animation/keyframe-animation.ts]: Just wondering - do you need both error callback and `.catch()`?

**Comment by vakrilov** [comment]
test

File: tns-core-modules/ui/animation/keyframe-animation.ts (hunk 1 above)
Comment (NathanWalker) [reply tns-core-modules/ui/animation/keyframe-animation.ts]: Although as @RoyiNamir points out I believe there are others cases that having both may provide utmost resiliency against.

**Pull request merged**